src/librustdoc/clean/types.rs

```diff
@@ -1094,10 +1081,17 @@ impl Attributes {
     /// The module can be different if this is a re-export with added documentation.
     crate fn collapsed_doc_value_by_module_level(&self) -> FxHashMap<Option<DefId>, String> {
         let mut ret = FxHashMap::default();
+        let last_string = match self.doc_strings.last() {
+            Some(string) => string,
+            None => return ret,
+        };
 
         for new_frag in self.doc_strings.iter() {
             let out = ret.entry(new_frag.parent_module).or_default();
             add_doc_fragment(out, new_frag);
+            if new_frag == last_string {
```

Comment:
This compares the whole string, not the index, which seems wrong: you could have two lines with the same docs. Can you use enumerate() and compare the indexes instead?

Reply:
Done. Alternatively, the conditional could be removed, and `pop` called on the entry for the last element after the loop. Would that be preferred? (That would make two instances of removing a conditional from a loop in the same PR!) It would probably only be better if 'self.doc_strings' has a lot of elements such that the cost of another hashing/retrieval operation is less expensive than numerous integer equality checks.

Reply:
Should there be a test to make sure this doesn't regress to what you described?

Reply:
If you can put a test together, that would be great! I don't think it's strictly necessary though.